source/real_locations.rst

```diff
@@ -0,0 +1,116 @@
+Checking the people in a location (SIR)
+=======================================
+
+What we were doing
+------------------
+
+While calculating the number of people who could potentially infect a person, we did the following:
+
+.. code-block:: scala
+
+    def computeInfectedFraction(node: Node, placeType: String, context: Context): Double = {
+    val totalNeighbourCount = node.getConnectionCount(node.getRelation[Person]().get)
+    if (totalNeighbourCount == 0)
+      return 0d
+    val infectedNeighbourCount = node.getConnectionCount(node.getRelation[Person]().get,
+      "infectionState" equ Infected).toDouble
+    infectedNeighbourCount / totalNeighbourCount.toDouble
+  }
+
+Let's take a closer look at the first line, how we calculate ``totalNeighbourCount``.
+
+.. code-block:: scala
+
+  val totalNeighbourCount = node.getConnectionCount(node.getRelation[Person]().get)
+
+Assume that the node was an ``Office``. In that case,
+
+.. code-block:: scala
+
+  node.getRelation[Person]().get
+
+returns the ``EMPLOYER_OF`` string. Now, ``totalNeighbourCount`` is counting the total number of nodes which the workplace node is the ``EMPLOYER_OF``.
+
+The problem arises when we introduce quarantining to our model. Someone who's infected may follow a different schedule, and stay at home. However, ``totalNeighbourCount`` *doesn't care* about the location a person has at a particular tick: all it does it count the number of people with the appropriate relations.
+
+As a result of this, quarantining individuals is far less effective than it should be. It still could have a minor effect (depending on your model), as someone who's infected does not go to their workplace. Regardless, that person still infects others currently at the workplace.
+
+.. note:: The same thing happens while calculating ``infectedNeighbourCount``. This has effects not just on the workplace, but on homes, hospitals, and other locations in your model too.
+
+Dealing with the problem
+------------------------
+
+There are two currently proposed methods:
+
+Using an attribute of the ``Person`` class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We can solve the problem by adding an attribute called ``currentLocation`` to the ``Person`` class.
+
+.. tip:: Remember to also initialize the attribute when created an instance of the class while extracting data from the CSV!
+
+After doing so, we need to add a behaviour which changes the ``currentLocation`` at every tick. We could do this by adding the following line of code in the ``Person`` class:
+
+.. code-block:: scala
+
+  private val checkCurrentLocation: Context => Unit = (context: Context) => {
+    val schedule = context.fetchScheduleFor(this).get
+    val locationNextTick: String = schedule.getForStep(context.getCurrentStep + 1)
+    if (currentLocation != locationNextTick) {
+      this.updateParam("currentLocation", locationNextTick)
+    }
+  }
+  addBehaviour(checkCurrentLocation)
```

Comment:
Split this code into two parts: just add the function like you have, and afterwards add a little text saying something like "once you've done this, you need to register this as a behaviour using ``addBehaviour(...)`` in the ``Person`` class. Behaviours are executed in the order they are defined, so it's best not to get people used to writing a function and immediately registering it as a behaviour underneath. In this case it doesn't do anything (since `updateParam` takes a whole tick to update, but still.